planner/core/rule_max_min_eliminate.go

```diff
@@ -222,22 +230,80 @@ func (a *maxMinEliminator) eliminateMaxMin(p LogicalPlan) LogicalPlan {
 		if len(agg.AggFuncs) == 1 {
 			// If there is only one aggFunc, we don't need to guarantee that the child of it is a data
 			// source, or whether the sort can be eliminated. This transformation won't be worse than previous.
-			return a.eliminateSingleMaxMin(agg)
+			return a.eliminateSingleMaxMin(agg, opt)
 		}
 		// If we have more than one aggFunc, we can eliminate this agg only if all of the aggFuncs can benefit from
 		// their column's index.
-		aggs, canEliminate := a.splitAggFuncAndCheckIndices(agg)
+		aggs, canEliminate := a.splitAggFuncAndCheckIndices(agg, opt)
 		if !canEliminate {
 			return agg
 		}
 		for i := range aggs {
-			aggs[i] = a.eliminateSingleMaxMin(aggs[i])
+			aggs[i] = a.eliminateSingleMaxMin(aggs[i], opt)
 		}
-		return a.composeAggsByInnerJoin(aggs)
+		return a.composeAggsByInnerJoin(agg, aggs, opt)
 	}
 	return p
 }
 
 func (*maxMinEliminator) name() string {
 	return "max_min_eliminate"
 }
+
+func appendEliminateSingleMaxMinTrace(agg *LogicalAggregation, sel *LogicalSelection, sort *LogicalSort, limit *LogicalLimit, opt *logicalOptimizeOp) {
+	action := func() string {
+		buffer := bytes.NewBufferString("")
+		if sel != nil {
+			buffer.WriteString(fmt.Sprintf("add selection[%v],", sel.ID()))
+		}
+		if sort != nil {
+			buffer.WriteString(fmt.Sprintf("add sort[%v],", sort.ID()))
+		}
+		buffer.WriteString(fmt.Sprintf("add limit[%v] during eliminate agg[%v] %s function", limit.ID(), agg.ID(), agg.AggFuncs[0].Name))
+		return buffer.String()
+	}()
+	reason := func() string {
+		buffer := bytes.NewBufferString(fmt.Sprintf("agg[%v] has only one function[%s] without group by", agg.ID(), agg.AggFuncs[0].Name))
+		if sel != nil {
+			buffer.WriteString(fmt.Sprintf(", the columns in agg[%v] shouldn't be NULL and needs to be filer NULL out", agg.ID()))
+		}
+		if sort != nil {
+			buffer.WriteString(fmt.Sprintf(", the columns in agg[%v] should be sorted", agg.ID()))
+		}
+		return buffer.String()
+	}()
+	opt.appendStepToCurrent(agg.ID(), agg.TP(), reason, action)
+}
+
+func appendEliminateMultiMinMaxTraceStep(originAgg *LogicalAggregation, aggs []*LogicalAggregation, joins []*LogicalJoin, opt *logicalOptimizeOp) {
+	action := func() string {
+		buffer := bytes.NewBufferString(fmt.Sprintf("agg[%v] splited into aggs[", originAgg.ID()))
+		for i, agg := range aggs {
+			if i > 0 {
+				buffer.WriteString(",")
+			}
+			buffer.WriteString(fmt.Sprintf("%v", agg.ID()))
+		}
+		buffer.WriteString("], and add joins[")
+		for i, join := range joins {
+			if i > 0 {
+				buffer.WriteString(",")
+			}
+			buffer.WriteString(fmt.Sprintf("%v", join.ID()))
+		}
+		buffer.WriteString(fmt.Sprintf("] as their parent during eliminate agg[%v] multi min/max functions", originAgg.ID()))
+		return buffer.String()
+	}()
+	reason := func() string {
+		buffer := bytes.NewBufferString("each column is sorted and has index in agg[")
```

Comment:
```suggestion
		buffer := bytes.NewBufferString("each column is sorted and can benefit from index/primary key in agg[")
```
Having an index is not enough to enable this optimization, so I think "benefit from" is more precise.